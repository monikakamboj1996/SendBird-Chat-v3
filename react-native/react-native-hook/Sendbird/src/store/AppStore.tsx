import AsyncStorage from '@react-native-async-storage/async-storage';
import { createStore, applyMiddleware } from 'redux';

import { persistStore, persistReducer } from 'redux-persist';
import thunk from 'redux-thunk';
import RootReducer from '../reducer/RootReducer';

// Middleware: Redux Persist Config
const persistConfig = {
  // Root
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,
};

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, RootReducer);

// Redux: Store
const store = createStore(persistedReducer, applyMiddleware(thunk));

// Middleware: Redux Persist Persister
let persistor = persistStore(store);
// Exports
export { store, persistor };
