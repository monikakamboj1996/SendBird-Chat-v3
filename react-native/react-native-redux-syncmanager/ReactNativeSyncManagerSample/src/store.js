import {createStore, compose, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from './reducer';

export const store = createStore(reducers, {}, compose(applyMiddleware(thunk)));
