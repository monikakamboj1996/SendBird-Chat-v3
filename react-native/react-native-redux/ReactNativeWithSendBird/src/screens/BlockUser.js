import React, { Component } from 'react';
import { View, FlatList, Text, TouchableHighlight, Alert } from 'react-native';
import { connect } from 'react-redux';
import { initBlockUser, getBlockUserList, onUnblockUserPress } from '../actions';
import { Button, Spinner, ListItem, Avatar } from '../components';
import { sbCreateBlockedUserListQuery } from '../sendbirdActions';

class BlockUser extends Component {
  static navigationOptions = ({ navigation }) => {
    const { params } = navigation.state;
    return {
      title: 'Block Users',
      headerLeft: (
        <Button
          containerViewStyle={{ marginLeft: 0, marginRight: 0 }}
          buttonStyle={{ paddingLeft: 14 }}
          icon={{
            name: 'chevron-left',
            type: 'font-awesome',
            color: '#7d62d9',
            size: 18
          }}
          backgroundColor="transparent"
          onPress={() => navigation.goBack()}
        />
      )
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      isLoading: false,
      blockedUserListQuery: null,
      list: []
    };
  }

  componentDidMount() {
    this._initBlockUser();
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { list, unblockedUserId } = this.props;

    if (list !== prevProps.list) {
      const newList = [...this.state.list, ...list];
      this.setState({
        isLoading: false,
        list: newList
      });
    }

    if (unblockedUserId !== prevProps.unblockedUserId) {
      const newList = this.state.list.filter(user => {
        return user.userId !== unblockedUserId;
      });
      this.setState({
        isLoading: false,
        list: newList
      });
    }
  }

  _initBlockUser = () => {
    this.props.initBlockUser();
    this._getBlockUserList(true);
  };

  _onUnblockUserPress = unblockUserId => {
    Alert.alert('User Unblock', 'Are you sure want to unblock user?', [
      { text: 'Cancel' },
      {
        text: 'OK',
        onPress: () => {
          this.setState({ isLoading: true });
          this.props.onUnblockUserPress(unblockUserId);
        }
      }
    ]);
  };

  _getBlockUserList = init => {
    if (!init && !this.props.blockedUserListQuery) {
      return;
    }
    this.setState({ isLoading: true }, () => {
      if (init) {
        const blockedUserListQuery = sbCreateBlockedUserListQuery();
        this.setState({ blockedUserListQuery }, () => {
          this.props.getBlockUserList(this.state.blockedUserListQuery);
        });
      } else {
        this.props.getBlockUserList(this.state.blockedUserListQuery);
      }
    });
  };

  _renderList = ({ item }) => {
    return (
      <ListItem
        key={item.userId}
        component={TouchableHighlight}
        containerStyle={{ backgroundColor: '#fff' }}
        avatar={
          <Avatar rounded source={item.profileUrl ? { uri: item.profileUrl } : require('../img/icon_sb_68.png')} />
        }
        title={item.nickname}
        titleStyle={{ fontWeight: '500', fontSize: 16 }}
        rightTitle={item.isOnline}
        rightTitleStyle={{ color: '#37b24d' }}
        rightIcon={<Text></Text>}
        onPress={() => this._onUnblockUserPress(item.userId)}
      />
    );
  };

  render() {
    return (
      <View>
        <Spinner visible={this.state.isLoading} />
        <FlatList
          renderItem={this._renderList}
          data={this.state.list}
          onEndReached={() => this._getBlockUserList(false)}
          onEndReachedThreshold={-100}
          keyExtractor={(item, index) => index.toString()}
        />
      </View>
    );
  }
}

function mapStateToProps({ blockUser }) {
  const { list, unblockedUserId } = blockUser;
  return { list, unblockedUserId };
}

export default connect(
  mapStateToProps,
  { initBlockUser, getBlockUserList, onUnblockUserPress }
)(BlockUser);
