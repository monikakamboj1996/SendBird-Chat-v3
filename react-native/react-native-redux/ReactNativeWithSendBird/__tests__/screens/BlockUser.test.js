import 'react-native';
import React from 'react';
import BlockUser from '../../src/screens/BlockUser';
import store from '../../src/store';
import renderer from 'react-test-renderer';

describe('screen/BlockUser', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BlockUser store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
