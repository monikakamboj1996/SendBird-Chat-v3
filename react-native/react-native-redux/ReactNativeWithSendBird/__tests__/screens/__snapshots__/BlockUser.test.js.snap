// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen/BlockUser renders correctly 1`] = `
<View>
  <RCTScrollView
    dataSource={
      ListViewDataSource {
        "items": 0,
      }
    }
    enableEmptySections={true}
    onEndReached={[Function]}
    onEndReachedThreshold={-100}
    renderRow={[Function]}
    renderScrollComponent={[Function]}
  >
    <View />
  </RCTScrollView>
</View>
`;
